x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorSyncInfo.java

```diff
@@ -105,18 +105,24 @@ public ConnectorSyncInfo(StreamInput in) throws IOException {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field(LAST_ACCESS_CONTROL_SYNC_ERROR.getPreferredName(), lastAccessControlSyncError);
         builder.field(LAST_ACCESS_CONTROL_SYNC_STATUS_FIELD.getPreferredName(), lastAccessControlSyncStatus);
-        builder.field(LAST_ACCESS_CONTROL_SYNC_SCHEDULED_AT_FIELD.getPreferredName(), lastAccessControlSyncScheduledAt);
+        builder.field(
+            LAST_ACCESS_CONTROL_SYNC_SCHEDULED_AT_FIELD.getPreferredName(),
+            ConnectorUtils.formatInstantToFrameworkString(lastAccessControlSyncScheduledAt)
+        );
         if (lastDeletedDocumentCount != null) {
             builder.field(LAST_DELETED_DOCUMENT_COUNT_FIELD.getPreferredName(), lastDeletedDocumentCount);
         }
-        builder.field(LAST_INCREMENTAL_SYNC_SCHEDULED_AT_FIELD.getPreferredName(), lastIncrementalSyncScheduledAt);
+        builder.field(
+            LAST_INCREMENTAL_SYNC_SCHEDULED_AT_FIELD.getPreferredName(),
+            ConnectorUtils.formatInstantToFrameworkString(lastIncrementalSyncScheduledAt)
```

Comment:
As I'm seeing now `formatInstantToFrameworkString` I'm wondering, if we should fix it the other way around. As the Connectors API guards everything around the protocol it feels cleaner, if the clients (like the framework) adapt to the API and not the other way around. In theory the connectors API should also be useable for other clients/schedulers.

Reply:
I agree that the issue should be fixed at the framework level, and I already filed an enhancement request https://github.com/elastic/connectors/issues/2067 to unify handling (parsing, representing) datetimes in connectors framework. 

While this PR doesn't solve the underlying root cause it mitigates the symptoms and allows us move forward with integration. Also, think about the connectors created in older releases, we need to be backward compatible with their representation with timestamps (or prepare a migration). Once the issue is solved in framework we can change the API formatting of timestamps.

Reply:
Thanks for the explanation, makes sense!